erpnext/accounts/doctype/promotional_scheme/promotional_scheme.py

```diff
@@ -73,42 +80,77 @@ def get_pricing_rules(doc, rules = {}):
 def _get_pricing_rules(doc, child_doc, discount_fields, rules = {}):
 	new_doc = []
 	args = get_args_for_pricing_rule(doc)
+	applicable_for = frappe.scrub(doc.get('applicable_for'))
 	for d in doc.get(child_doc):
 		if d.name in rules:
-			pr = frappe.get_doc('Pricing Rule', rules.get(d.name))
+			for applicable_for_value in args.get(applicable_for):
+				temp_args = args.copy()
+				docname = frappe.get_all(
+					'Pricing Rule',
+					fields = ["promotional_scheme_id", "name", applicable_for],
+					filters = {
+						'promotional_scheme_id': d.name,
+						applicable_for: applicable_for_value
+					}
+				)
+					
+				if docname:
+					pr = frappe.get_doc('Pricing Rule', docname[0].get('name'))
+					temp_args[applicable_for] = applicable_for_value
+					pr = set_args(temp_args, pr, doc, child_doc, discount_fields, d)
+				else:
+					pr = frappe.new_doc("Pricing Rule")
+					pr.title = make_autoname("{0}/.####".format(doc.name))
```

Comment:
why are we forcing this naming as the title here ? Shouldn't it automatically pick this up from whatever is configured ?